package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,587 @@
+        # 'position', 'velocity', or 'force' groups in the file
+        self._has = {name: name in self._particle_group for
+                     name in ('position', 'velocity', 'force')}
+
+        # Gets n_atoms from first available group
+        for name, value in self._has.items():
+            if value:
+                self.n_atoms = self._particle_group[name]['value'].shape[1]
+                break
+        else:
+            raise NoDataError("Provide at least a position, velocity"
+                              " or force group in the h5md file.")
+
+        self.ts = self._Timestep(self.n_atoms,
+                                 positions=self.has_positions,
+                                 velocities=self.has_velocities,
+                                 forces=self.has_forces,
+                                 **self._ts_kwargs)
+
+        self.units = {'time': None,
+                      'length': None,
+                      'velocity': None,
+                      'force': None}
+        self._set_translated_units()  # fills units dictionary
+        self._read_next_timestep()
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Reader read *thing*"""
+        # nb, filename strings can still get passed through if
+        # format='H5MD' is used
+        return HAS_H5PY and isinstance(thing, h5py.File)
+
+    def open_trajectory(self):
+        """opens the trajectory file using h5py library"""
+        self._frame = -1
+        if isinstance(self.filename, h5py.File):
+            self._file = self.filename
+            self._driver = self._file.driver
+        else:
+            if self._comm is not None:
+                # can only pass comm argument to h5py.File if driver='mpio'
+                assert self._driver == 'mpio'
+                self._file = h5py.File(self.filename, 'r',
+                                       driver=self._driver,
+                                       comm=self._comm)
+            elif self._driver is not None:
+                self._file = h5py.File(self.filename, 'r', driver=self._driver)
+            else:
+                self._file = h5py.File(self.filename, 'r')
+        # pulls first key out of 'particles'
+        # allows for arbitrary name of group1 in 'particles'
+        self._particle_group = self._file['particles'][
+                               list(self._file['particles'])[0]]
+
+    def close(self):
+        """close reader"""
+        if self._driver != 'mpio':
+            self._file.close()
+
+    def _reopen(self):
+        """reopen trajectory
+
+        Note
+        ----
+        This method does close the and reopen the file like most other
+        coordinate readers, as this would cause problems with reopening
+        the file with the ``driver`` and ``comm`` arguments for parallel
+        reads. Instead, it rewinds the trajectory back to the first timstep
+        if 'mpio' is passed for the driver.
```

Comment:
Text needs some corrections, maybe
```suggestion
        If the `driver` and `comm` arguments were used to open the
        hdf5 file (specifically, ``driver="mpio"``) then this method 
        does *not* close and open the file like most readers because 
        the information about the MPI communicator would be lost; instead
        it rewinds the trajectory back to the first timstep.
```